superset/connectors/base/models.py

```diff
@@ -312,9 +312,9 @@ def data_for_slices(  # pylint: disable=too-many-locals
                 for metric in utils.get_iterable(form_data.get(metric_param) or []):
                     metric_names.add(utils.get_metric_name(metric))
                     if utils.is_adhoc_metric(metric):
-                        column_names.add(
-                            (metric.get("column") or {}).get("column_name")
-                        )
+                        column = metric.get("column") or {}
+                        if hasattr(column, "get") and "column_name" in column:
+                            column_names.add(column.get("column_name"))
```

Comment:
Looking at the `AdhocMetric` class, it appears `column` is `Optional[AdhocMetricColumn]`, which in turn is a `TypedDict`. So `hasattr(column, "get")` will always be true based on line 315. So I believe this can be simplified to
```suggestion
                        if column_name := column.get("column_name"):
                            column_names.add(column_name)
```